module/VuFind/src/VuFind/Cover/Router.php

```diff
@@ -48,30 +49,17 @@ class Router implements \Laminas\Log\LoggerAwareInterface
 {
     use \VuFind\Log\LoggerAwareTrait;
 
-    /**
-     * Base URL for dynamic cover images.
-     *
-     * @var string
-     */
-    protected $dynamicUrl;
-
-    /**
-     * Cover loader
-     *
-     * @var CoverLoader
-     */
-    protected $coverLoader;
-
     /**
      * Constructor
      *
      * @param string      $url         Base URL for dynamic cover images.
      * @param CoverLoader $coverLoader Cover loader
      */
-    public function __construct($url, CoverLoader $coverLoader)
-    {
-        $this->dynamicUrl = $url;
-        $this->coverLoader = $coverLoader;
+    public function __construct(
+        protected string $dynamicUrl,
+        protected CoverLoader $coverLoader,
+        protected Config $config
```

Comment:
laminas-config has just been abandoned; I'm beginning work to remove it from the project. I'd suggest using a plain array here, and using `->toArray` in the factory, so we don't introduce a new dependency on the abandoned project.

Reply:
It might also make sense to default this to an empty array so that you don't have to modify tests when there is no configuration being tested.

Reply:
Done.

Are there any problems with replacing laminas-config with arrays that would be good to keep in mind?

Reply:
My plan is to create a `VuFind\Config\Config` that provides backward-compatibility with `Laminas\Config\Config` as a first phase. We might decide to keep that in the long term, but I think it's more likely that we'll quickly deprecate it and try to phase it out in favor of arrays. Really, the Laminas Config object was mainly useful for accessing deep configs in older versions of PHP prior to the introduction of null coalescing. Now that we have `??`, it's pretty easy to access deeply-nested array properties as long as you provide a default to fall back on.

Reply:
I see, thanks